budgetportal/tests/test_prov_infra_projects.py

```diff
@@ -1,247 +1,959 @@
 import os
+import time
+from datetime import date
 
+from django.core.files import File
+
+from budgetportal.models import (
+    FinancialYear,
+    IRMSnapshot,
+    ProvInfraProject,
+    ProvInfraProjectSnapshot,
+    Quarter,
+)
+from budgetportal.search_indexes import ProvInfraProjectIndex
+from budgetportal.tests.helpers import BaseSeleniumTestCase
 from django.urls import reverse
 from rest_framework import status
-from rest_framework.test import APITestCase
-from django.test import TestCase
-from django.contrib.auth.models import User
-from allauth.account.models import EmailAddress
-from selenium.webdriver.support.select import Select
-from tablib import Dataset
-import unittest
-
-from budgetportal.models import FinancialYear, ProvInfraProject
-from budgetportal import irm_preprocessor
-from budgetportal.tests.helpers import BaseSeleniumTestCase
+from rest_framework.test import APITransactionTestCase
 
 USERNAME = "testuser"
 EMAIL = "testuser@domain.com"
 PASSWORD = "12345"
 
 
-class ProvInfraProjectSeleniumTestCase(BaseSeleniumTestCase):
+class ProvInfraProjectIRMSnapshotTestCase(APITransactionTestCase):
     def setUp(self):
-        user = User.objects.create_user(
-            username=USERNAME,
-            password=PASSWORD,
-            is_staff=True,
-            is_superuser=True,
-            is_active=True,
-        )
-        EmailAddress.objects.create(user=user, email=EMAIL, verified=True)
-        self.path = os.path.dirname(__file__)
-        self.financial_year = FinancialYear.objects.create(slug="2019-20")
-        self.timeout = 10
+        file_path = os.path.abspath(
+            ("budgetportal/tests/test_data/test_import_prov_infra_projects-update.xlsx")
+        )
+        self.file = File(open(file_path))
+        self.financial_year = FinancialYear.objects.create(slug="2030-31")
+        self.quarter = Quarter.objects.create(number=1)
+        self.date = date(year=2050, month=1, day=1)
+        IRMSnapshot.objects.create(
+            financial_year=self.financial_year,
+            quarter=self.quarter,
+            date_taken=self.date,
+            file=self.file,
+        )
+        self.url = reverse("provincial-infrastructure-project-api-list")
+
+    def test_import_irm_snapshot(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Check that 3 projects are indexed and searchable
+        results = response.data["results"]
+        num_of_results = len(results)
+        self.assertEqual(num_of_results, 3)
```

Comment:
Please create the snapshot inside the test case and assert that there are zero projects before creating it?

Reply:
I moved the create to the test body.

https://github.com/vulekamali/datamanager/pull/213/commits/8e32bab500c8434735e73eea5219aaefc3969a39